resources/lib/cbc.py

```diff
@@ -29,96 +61,195 @@ def __init__(self):
         if session_cookies is not None:
             self.session.cookies = session_cookies
 
-    def authorize(self, username=None, password=None, callback=None):
-        """Authorize for video playback."""
-        token = self.radius_login(username, password)
-        if callback is not None:
-            callback(25)
-        if token is None:
-            log('Radius Login failed', True)
+    @staticmethod
+    def azure_authorize_authorize(sess: requests.Session):
+        """
+        Make the first authorization call.
+        @param sess A requests session
+        """
+        nonce= str(uuid4())
+        guid = str(uuid4())
+        state_str = f'{guid}|{{"action":"login","returnUrl":"/","fromSubscription":false}}'.encode()
+        state = b64encode(state_str).decode('ascii')
+        params = {
+            'client_id': 'fc05b0ee-3865-4400-a3cc-3da82c330c23',
+            'nonce': nonce,
+            'redirect_uri': 'https://gem.cbc.ca/auth-changed',
+            'scope': SCOPES,
+            'response_type': 'id_token token',
+            'response_mode': 'fragment',
+            'state': state,
+            'state_value': state,
+            'ui_locales': 'en',
+        }
+        resp = sess.get(AUTHORIZE_LOGIN, params=params)
+        if resp.status_code != 200:
+            log('Call to authorize fails', True)
             return False
 
-        jwt = self.radius_jwt(token)
-        if callback is not None:
-            callback(50)
-        if jwt is None:
-            log('Radius JWT retrieval failed', True)
+        if not 'x-ms-gateway-requestid' in resp.headers:
+            log('authorize authorize response had no x-ms-gateway-requestid header')
```

Comment:
"`authorize authorize`" typo?

Reply:
Yes, do you insist its fixed before merging?

Reply:
oh, thought this was the PR i submitted (hence my question). Anyway - thanks for pointing it out.

Reply:
not a blocker for sure.. just a point out.. would have approved, but since no running automated test to assert with, not sure how to proceed. I did  went through the rest of the code and nothing else stood out. great job! is it enough for me to just approve?

Reply:
no - this is just a PR i created for myself. I'll merge it when the PR into the real addons repo merges.

Reply:
or six months later.